submissions/crealime/oop-exercise/index.js

```diff
@@ -0,0 +1,143 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/crealime/a-tiny-JS-world
+   Web app: https://crealime.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+
+function getRandomId() {
+  return  new Date().getTime() + '_' + Math.floor(10000 + Math.random() * (99999 + 1 - 10000))
+}
+
+class Inhabitant {
+  constructor(species, name, gender, saying) {
+    this.id = getRandomId()
+    this.species = species
+    this.name = name
+    this.gender = gender
+    this.saying = saying
+    this.friends = []
+  }
+
+  getFullName() {
+    return this.name
+  }
+}
+
+class Dog extends Inhabitant {
+  constructor(name, gender, legs) {
+    super('dog', name, gender, 'Woof-woof!')
+    this.legs = legs
+  }
+}
+
+class Cat extends Inhabitant {
+  constructor(name, gender, legs) {
+    super('cat', name, gender, Cat.say())
+    this.legs = legs
+  }
+
+  static say() {
+    return 'Meow-meow!'
+  }
+}
+
+class Human extends Inhabitant {
+  constructor(name, surname, gender, legs, hands, saying) {
+    super('human', name, gender, saying)
+    this.surname = surname
+    this.legs = legs
+    this.hands = hands
+  }
+
+  getFullName() {
+    return `${super.getFullName()} ${this.surname}`
+  }
+}
+
+class CatWoman extends Inhabitant {
+  constructor(name, surname) {
+    super('cat-woman', name, 'female', Cat.say())
+    this.surname = surname
+    this.legs = 2
+    this.hands = 2
+  }
+
+  getFullName() {
+    return `${this.name} ${this.surname}`
+  }
+}
+
+const dog = new Dog('Marley', 'male', 4)
+const cat = new Cat('Sophie', 'female', 4)
+const man = new Human('Harry', 'Taylor', 'male', 2, 2, 'Hello everyone!')
+const woman = new Human('Emily', 'Brown', 'female', 2, 2, 'Hi all!')
+const pirate = new Human('John', 'Silver', 'male', 1, 2, 'Give no quarter!')
+const catWoman = new CatWoman('Selina', 'Kyle')
+
+dog.friends.push(cat, man)
+cat.friends.push(dog, woman, catWoman)
+man.friends.push(dog, woman)
+woman.friends.push(cat, man)
+catWoman.friends.push(cat)
+
+const secretSociety = []
+const allInhabitant = [dog, cat, man, woman, catWoman, pirate]
+
+function addToSociety(society, member) {
+  society.push(member.id)
+}
+
+addToSociety(secretSociety, catWoman)
+addToSociety(secretSociety, pirate)
+
+function isInSociety(society, members) {
+  return members.filter(member => society.includes(member.id)).map(member => member.name)
+}
```

Comment:
Looks like a secret society deserves to be a class. Just for the sake of scoping and encapsulation of storage and behaviour.
Optional: think of Singleton pattern.

Reply:
Done.